### PR TITLE
Adds support for calculating maximum CPU

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Similarly to how we specify a maximum amount of memory (to fit within the nodes of a cluster), this PR adds support to specify a maximum amount of CPU.

e.g:
```
$ kubectl get nodes ip-10-0-5-109.eu-central-1.compute.internal -o yaml | grep -A 2 allocatable
        f:allocatable:
          .: {}
          f:attachable-volumes-aws-ebs: {}
--
  allocatable:
    attachable-volumes-aws-ebs: "25"
    cpu: "4"
```
```
$ kubectl -n=gorilla-prometheus get vpa prometheus -o yaml | grep -A 2 maxAllowed
      maxAllowed:
        cpu: "3"
        memory: 14325966Ki
```

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: service/controller/resource/verticalpodautoscaler/resource.go [hunk 1]
```diff
@@ -135,6 +140,39 @@ func (r *Resource) getObject(ctx context.Context, v interface{}) (*vpa_types.Ver
...
+		}
+
+		for _, n := range nodes.Items[1:] {
+			s, ok := n.Status.Allocatable[v1.ResourceCPU]
+			if ok && nodeCpu.Cmp(s) == -1 {
+				nodeCpu = &s
+			}
+		}
+	}
+	if nodeCpu == nil || nodeCpu.IsZero() {
```
Comment (JosephSalisbury) [inline service/controller/resource/verticalpodautoscaler/resource.go:164]: If `len(nodes.Items) == 0`, then `nodeCpu` could be nil, which would panic is we just called `IsZero` on. Fixed below as well.

**Comment by JosephSalisbury** [comment]
Bam tests fixed

File: service/controller/resource/verticalpodautoscaler/resource.go (hunk 1 above)
Comment (QuentinBisson) [reply service/controller/resource/verticalpodautoscaler/resource.go:164]: Nice

**Review by QuentinBisson — APPROVED** [review]
LGTM if tested

**Pull request merged**